src/VisualStudio/Core/Def/Implementation/Workspace/Esent/EsentPersistentStorage.cs

```diff
@@ -274,12 +274,10 @@ private bool TryGetUniqueId(string value, bool fileCheck, out int id)
 
             try
             {
-                id = _nameTableCache.GetOrAdd(value, v =>
-                {
-                    // okay, get one from esent
-                    var uniqueIdValue = fileCheck ? FilePathUtilities.GetRelativePath(Path.GetDirectoryName(SolutionFilePath), v) : v;
-                    return _esentStorage.GetUniqueId(uniqueIdValue);
-                });
+                var uniqueIdValue = fileCheck ? FilePathUtilities.GetRelativePath(Path.GetDirectoryName(SolutionFilePath), value) : value;
+                id = _nameTableCache.GetOrAdd(value, _esentStorage.GetUniqueId(uniqueIdValue));
```

Comment:
cost should be same. that line only runs when key is not in the dictionary already. so only chance it would run multiple time for same key is race case. and in race could, previous implementation (concurrent dictionary) also runs it multiple times.